### PR TITLE
parser: cleanup using token.is_next_to()

### PR DESCRIPTION
**Pull request by yuyi98**
This PR cleanup using `token.is_next_to()`.

### CODE REVIEW

File: vlib/v/parser/parser.v [hunk 1]
```diff
@@ -2240,7 +2240,7 @@ fn (mut p Parser) parse_multi_expr(is_top_level bool) ast.Stmt {
 }
 
 fn (mut p Parser) is_following_concrete_types() bool {
-	if !(p.tok.kind == .lsbr && p.tok.pos - p.prev_tok.pos == p.prev_tok.len) {
+	if !(p.tok.kind in [.lt, .lsbr] && p.tok.is_next_to(p.prev_tok)) {
```
[the remark this answers is not in the document]
Comment (spytheman) [reply vlib/v/parser/parser.v:2243]: does it break some test case?

[the remark this answers is not in the document]
Comment (yuyi98) [reply vlib/v/parser/parser.v:2243]: Foo<int> or Foo[int]

[the remark this answers is not in the document]
Comment (yuyi98) [reply vlib/v/parser/parser.v:2243]: I have tested all the test cases.

[the remark this answers is not in the document]
Comment (spytheman) [reply vlib/v/parser/parser.v:2243]: I forgot that the parser still supports the old `<X>` syntax too.
Thanks.

**Pull request merged**